Modules/Messages/src/Messages.Common/Range.cs

```diff
@@ -10,17 +10,17 @@ public Range(T from, T to)
         To = to;
     }
 
-    public T From { get; set; }
-    public T To { get; set; }
+    public T? From { get; set; }
+    public T? To { get; set; }
 
     public bool HasFrom()
     {
-        return !From.Equals(default(T));
+        return From?.Equals(default(T)) ?? false;
     }
```

Comment:
As it is right now I **think** that when `From` is `null` the `From?` will force the flow to return `false` after `??`, and if it is not it will be compared with `default(T)` which may also be null, i.e. it will always be false.